truss/templates/server/common/truss_server.py

```diff
@@ -1,248 +1,169 @@
+import json
+import logging
+from typing import Dict, Optional, Union
+
+import kserve
+import kserve.errors as errors
+import numpy as np
+from common.serialization import (
+    DeepNumpyEncoder,
     truss_msgpack_deserialize,
     truss_msgpack_serialize,
 )
-from common.util import (  # noqa: E402
-    assign_request_to_inputs_instances_after_validation,
-)
-from kfserving import KFModel
-from kfserving.handlers.http import HTTPHandler  # noqa: E402
-from kfserving.kfserver import HealthHandler, KFServer, ListHandler, LivenessHandler
+from common.util import assign_request_to_inputs_instances_after_validation
+from fastapi import Depends, FastAPI, Request
+from fastapi.responses import ORJSONResponse
+from fastapi.routing import APIRoute as FastAPIRoute
+from kserve.errors import InvalidInput
+from kserve.handlers import DataPlane, ModelRepositoryExtension, V1Endpoints
 from model_wrapper import ModelWrapper
-from pythonjsonlogger import jsonlogger  # noqa: E402
-from tornado.httpserver import HTTPServer
-from tornado.ioloop import IOLoop
-
-ensure_kfserving_installed()
+from starlette.responses import Response
 
 logger = logging.getLogger(__name__)
 
 
-def _configure_logging():
-    json_log_handler = logging.StreamHandler()
-    json_log_handler.setFormatter(
-        jsonlogger.JsonFormatter("%(asctime)s %(levelname)s %(message)s")
-    )
-    logger = logging.getLogger()
-    for handler in logger.handlers:
-        logger.removeHandler(handler)
-    logger.addHandler(json_log_handler)
+async def parse_body(request: Request):
+    data: bytes = await request.body()
+    return data
+
+
+class BasetenEndpoints(V1Endpoints):
+    def __init__(
+        self,
+        dataplane: DataPlane,
+        model_repository_extension: Optional[ModelRepositoryExtension] = None,
+    ):
+        super().__init__(dataplane, model_repository_extension)
+
+    @staticmethod
+    def check_healthy(model: ModelWrapper):
+        model.start_load()
+
+        if model.load_failed():
+            raise errors.InferenceError("Model load failed")
+
+        if not model.ready:
+            raise errors.ModelNotReady(model.name)
 
+    async def model_ready(self, model_name: str) -> Dict[str, Union[str, bool]]:
+        model: ModelWrapper = self.dataplane.get_model_from_registry(model_name)
+
+        self.check_healthy(model)
+
+        return {}
+
+    def predict(
+        self, model_name: str, request: Request, body_raw: bytes = Depends(parse_body)
+    ) -> Response:
+        model: ModelWrapper = self.dataplane.get_model_from_registry(model_name)
+
+        self.check_healthy(model)
+
+        body: dict
+        if self.is_binary(request):
+            body = truss_msgpack_deserialize(body_raw)
+        else:
+            body = json.loads(body_raw)
+
+        body = assign_request_to_inputs_instances_after_validation(body)
 
-class TrussHTTPBinaryHandler(HTTPHandler):
-    def validate(self, request):
         if (
```

Comment:
Should we keep this refactored into separate method like it was before?

Reply:
This was a notion from Tornado that doesn't exist anymore in this new version of kserve. I rolled these into the predict method as it was only a few lines. We can break out to its own method, but it almost made the code a bit more confusing.

Reply:
> This was a notion from Tornado that doesn't exist anymore in this new version of kserve

I take this back, I found this within model, but it's not part of the endpoints anymore. Let me see if I can move this logic back into the model instead.

Reply:
I centralized this logic back in the model, which seems to be cleaner now. Will be in next commit.  Thanks for the push there and it definitely is cleaner! kserve change had me thinking they just threw it out.